imap_processing/tests/mag/test_mag_l1b.py

```diff
@@ -0,0 +1,77 @@
+from pathlib import Path
+
+import numpy as np
+import pytest
+import xarray as xr
+
+from imap_processing.cdf.utils import load_cdf, write_cdf
+from imap_processing.mag.l1b.mag_l1b import mag_l1b, mag_l1b_processing
+
+
+@pytest.fixture(scope="module")
+def mag_l1a_dataset():
+    epoch = xr.DataArray(np.arange(20), name="epoch", dims=["epoch"])
+    direction = xr.DataArray(np.arange(4), name="direction", dims=["direction"])
+    vectors = xr.DataArray(
+        np.zeros((20, 4)),
+        dims=["epoch", "direction"],
+        coords={"epoch": epoch, "direction": direction},
+    )
+
+    vectors[0, :] = np.array([1, 1, 1, 0])
+
+    output_dataset = xr.Dataset(
+        coords={"epoch": epoch, "direction": direction},
+    )
+    output_dataset["vectors"] = vectors
+
+    return output_dataset
+
+
+def test_mag_processing(mag_l1a_dataset):
+    mag_l1a_dataset.attrs["Logical_source"] = ["imap_mag_l1a_norm-mago"]
+
+    mag_l1b = mag_l1b_processing(mag_l1a_dataset)
+
+    assert np.allclose(
```

Comment:
Use the numpy testing helpers if possible. This will handle shape comparisons as well.
https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html
```suggestion
    np.testing.assert_allclose(
```